<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="0" total="8" passed="8">
    <reporter-output>
    </reporter-output>
    <suite name="Identity-connector-session-validation-test-suite" duration-ms="508" started-at="2018-05-12T04:53:42Z"
           finished-at="2018-05-12T04:53:43Z">
        <groups>
        </groups>
        <test name="session-validation-authenticator-all" duration-ms="508" started-at="2018-05-12T04:53:42Z"
              finished-at="2018-05-12T04:53:43Z">
            <class name="org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest">
                <test-method status="PASS"
                             signature="setup()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="setup" is-config="true" duration-ms="66" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- setup -->
                <test-method status="PASS"
                             signature="testGetFriendlyName()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="testGetFriendlyName" duration-ms="1" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testGetFriendlyName -->
                <test-method status="PASS"
                             signature="testGetName()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="testGetName" duration-ms="0" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testGetName -->
                <test-method status="PASS"
                             signature="testProcessInitialRequest()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="testProcessInitialRequest" duration-ms="4" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <exception
                            class="org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException">
                        <message>
                            <![CDATA[Authentication failed!. Failed to identify the user]]>
                        </message>
                        <full-stacktrace>
                            <![CDATA[org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException: Authentication failed!. Failed to identify the user
	at org.wso2.carbon.identity.application.authenticator.sessionauth.SessionCountAuthenticator.initiateAuthRequest(SessionCountAuthenticator.java:104)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.SessionCountAuthenticator.process(SessionCountAuthenticator.java:84)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest.testProcessInitialRequest(AuthenticatorTest.java:113)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:295)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:90)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
]]>
                        </full-stacktrace>
                    </exception> <!-- org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException -->
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testProcessInitialRequest -->
                <test-method status="PASS"
                             signature="testProcessInitialRequestWithValidUser()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="testProcessInitialRequestWithValidUser" duration-ms="27"
                             started-at="2018-05-12T10:23:43Z" finished-at="2018-05-12T10:23:43Z">
                    <exception class="java.lang.NullPointerException">
                        <full-stacktrace>
                            <![CDATA[java.lang.NullPointerException
	at java.util.Hashtable.put(Hashtable.java:459)
	at java.util.Properties.setProperty(Properties.java:166)
	at java.lang.System.setProperty(System.java:796)
	at org.wso2.carbon.utils.CarbonUtils.getCarbonHome(CarbonUtils.java:253)
	at org.wso2.carbon.utils.CarbonUtils.getCarbonConfigDirPath(CarbonUtils.java:300)
	at org.wso2.carbon.identity.core.util.IdentityUtil.getIdentityConfigDirPath(IdentityUtil.java:346)
	at org.wso2.carbon.identity.application.authentication.framework.config.builder.FileBasedConfigurationBuilder.buildConfiguration(FileBasedConfigurationBuilder.java:145)
	at org.wso2.carbon.identity.application.authentication.framework.config.builder.FileBasedConfigurationBuilder.<init>(FileBasedConfigurationBuilder.java:136)
	at org.wso2.carbon.identity.application.authentication.framework.config.builder.FileBasedConfigurationBuilder.getInstance(FileBasedConfigurationBuilder.java:100)
	at org.wso2.carbon.identity.application.authentication.framework.config.ConfigurationFacade.<init>(ConfigurationFacade.java:50)
	at org.wso2.carbon.identity.application.authentication.framework.config.ConfigurationFacade.getInstance(ConfigurationFacade.java:60)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.util.AuthenticatorUtil.getLoginPageURL(AuthenticatorUtil.java:173)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.SessionCountAuthenticator.initiateAuthRequest(SessionCountAuthenticator.java:109)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.SessionCountAuthenticator.process(SessionCountAuthenticator.java:84)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest.testProcessInitialRequestWithValidUser(AuthenticatorTest.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:295)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:90)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
]]>
                        </full-stacktrace>
                    </exception> <!-- java.lang.NullPointerException -->
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testProcessInitialRequestWithValidUser -->
                <test-method status="PASS"
                             signature="testProcessLogout()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest@7a1ebcd8]"
                             name="testProcessLogout" duration-ms="1" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testProcessLogout -->
            </class> <!-- org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorTest -->
            <class name="org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest">
                <test-method status="PASS"
                             signature="setup()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest@34cd072c]"
                             name="setup" is-config="true" duration-ms="144" started-at="2018-05-12T10:23:42Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- setup -->
                <test-method status="PASS"
                             signature="testGetQuery()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest@34cd072c]"
                             name="testGetQuery" duration-ms="4" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testGetQuery -->
                <test-method status="PASS"
                             signature="testGetSessionDetails()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest@34cd072c]"
                             name="testGetSessionDetails" duration-ms="246" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <exception class="java.lang.NullPointerException">
                        <full-stacktrace>
                            <![CDATA[java.lang.NullPointerException
	at java.net.URI$Parser.parse(URI.java:3042)
	at java.net.URI.<init>(URI.java:588)
	at java.net.URI.create(URI.java:850)
	at org.apache.http.client.methods.HttpPost.<init>(HttpPost.java:76)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.util.AuthenticatorUtil.createHttpRequest(AuthenticatorUtil.java:109)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.util.AuthenticatorUtil.getSessionDetails(AuthenticatorUtil.java:68)
	at org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest.testGetSessionDetails(AuthenticatorUtilTest.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:295)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:90)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
]]>
                        </full-stacktrace>
                    </exception> <!-- java.lang.NullPointerException -->
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testGetSessionDetails -->
                <test-method status="PASS"
                             signature="testSetAuthorizationHeader()[pri:0, instance:org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest@34cd072c]"
                             name="testSetAuthorizationHeader" duration-ms="4" started-at="2018-05-12T10:23:43Z"
                             finished-at="2018-05-12T10:23:43Z">
                    <reporter-output>
                    </reporter-output>
                </test-method> <!-- testSetAuthorizationHeader -->
            </class> <!-- org.wso2.carbon.identity.application.authenticator.sessionauth.test.AuthenticatorUtilTest -->
        </test> <!-- session-validation-authenticator-all -->
    </suite> <!-- Identity-connector-session-validation-test-suite -->
</testng-results>
